camera_control/camera_control/nodes/camera_control_node.py

```diff
@@ -0,0 +1,264 @@
+            "camera_control/control_laser_pub",
+            self.control_laser_pub,
+        )
+        self.laser_pub_bool = False
+
+        self.control_runner_pub_srv = self.create_service(
+            SendEnable,
+            "camera_control/control_runner_pub",
+            self.control_runner_pub,
+        )
+        self.runner_pub_control = False
+
+        self.camera = RealSense(self.logger, self.rgb_size, self.depth_size)
+        self.background_frame = None
+        self.initialize()
+
+    def initialize(self):
+        # Setup yolo model
+        include_dir = os.path.join(
+            get_package_share_directory("camera_control"), "include"
+        )
+        self.model = YOLO(os.path.join(include_dir, "RunnerSegModel.pt"))
+
+        self.camera.initialize()
+        self.initialize_recording()
+
+    def initialize_recording(self):
+        # handle default log location
+        ts = time.time()
+        datetime_obj = datetime.fromtimestamp(ts)
+        datetime_string = datetime_obj.strftime("%Y%m%d%H%M%S")
+        if self.rec_video_frame:
+            video_name = f"{datetime_string}.avi"
+            rec_name = os.path.join(self.video_dir, video_name)
+            self.rec = cv2.VideoWriter(
+                rec_name, 0, 1 / self.frame_period, (self.rgb_size[0], self.rgb_size[1])
+            )
+        if self.rec_debug_frame:
+            debug_video_name = f"{datetime_string}_debug.avi"
+            debug_rec_name = os.path.join(self.debug_video_dir, debug_video_name)
+            self.rec_debug = cv2.VideoWriter(
+                debug_rec_name,
+                0,
+                1 / self.frame_period,
+                (self.rgb_size[0], self.rgb_size[1]),
+            )
+
+    def frame_callback(self):
+        frames = self.camera.get_frames()
+        self.curr_frames = frames
+        if not frames:
+            return
+
+        # This is still using a realsense frame concept, for better multicamera
+        # probability, all realsense based frame controls should move into the
+        # realsense module.
+        frame_ts = frames["color"].get_timestamp()
+        # convert from ms to seconds
+        frame_ts = frame_ts / 1000
+        self.logger.debug(
+            f"Publishing frame ts: {frame_ts}, current time:{time.time()}"
+        )
+
+        curr_image = np.asanyarray(frames["color"].get_data())
+        self.rec.write(curr_image)
+
+        laser_point_list = []
+        runner_point_list = []
+
+        if self.laser_pub_control:
+            laser_point_list = cv_utils.detect_laser(frames)
+            laser_msg = self.create_pos_data_msg(laser_point_list, frames)
+            self.laser_pos_publisher.publish(laser_msg)
+        else:
+            # If not publishing laser information save the background image
+            self.background_image = curr_image
+        if self.runner_pub_control:
+            runner_point_list = cv_utils.detect_runners(frames, self.background_image)
```

Comment:
this doesn't seem to match the params that cv_utils.detect_runners expects

Reply:
Yeah, I had some errors in the needed variables, fixed them all up